python/lsst/pipe/supertask/super_task.py

```diff
@@ -223,103 +119,69 @@ def completed(self, completed):
         Sets the status of (Super)Task. The Task should be able to report whether it ran
         successfully or not, this is useful for workflow and freeze-dry runs or to make checkpoints.
 
-        :param completed: Current status, usually this is done in post_run()
+        :param completed: New task completion status.
         """
         self._completed = completed
 
-    def pre_run(self, *args, **kwargs):
-        """
-        Prerun method, which given the decorator @wraprun will run before run()
-        This method can be used to prepare the data or logs before running the task
-        """
-        pass
-
-    def post_run(self, *args, **kwargs):
-        """
-        Postrun method, this hook method can be used to define the completion of a task and to
-        gather information about the run()
-        """
-        # By default, if used, this methods set the completion to be true.
-        if args is None and kwargs in None:
-            self.completed = True
-
     def run(self, *args, **kwargs):
         """
-        This function is the one that actually operates on teh data (exposed by execute) and usually
-        returning a Struct with the collected results
+        Run task algorithm on in-memory data.
```

Comment:
It's up to the task writer to define what arguments `run()` accepts. What defined in this class does not really matter, each subclass will have to define its own run method with specific argument list (e.g. 'def run(self, imgDataRef, doSomethingFancy=Flase)`)